src/sentry/dynamic_sampling/rules_generator.py

```diff
@@ -19,6 +21,57 @@
 )
 from sentry.models import Project
 
+
+class DSRulesLogger:
+    def __init__(self, rules: List[Tuple[RuleType, Union[BaseRule, ReleaseRule]]]):
+        self.logger = logging.getLogger("dynamic_sampling.rules")
+        self.rules = rules
+
+    def log_rules(self) -> None:
+        try:
+            self.logger.info(
+                "rules_generator.generate_rules",
+                extra={
+                    "rules": self._format_rules(),
+                    # We set the current date as creation timestamp, however, this is not indicating that Relay
+                    # did apply the rules at this time as there will be a non-deterministic delay before that happens.
+                    "creation_timestamp": datetime.now(tz=pytz.utc),
+                },
+            )
+        except Exception as e:
+            # If there is any problem while generating the log message, we just silently fail and notify the error to
+            # Sentry.
+            sentry_sdk.capture_exception(e)
+
+    def _format_rules(self) -> List[Dict[str, Union[List[str], str, float, None]]]:
+        formatted_rules = []
+
+        for rule_type, rule in self.rules:
+            formatted_rules.append(
+                {
+                    "type": rule_type.value,
+                    "id": rule["id"],
+                    "sample_rate": rule["sampleRate"],
+                    **self._extract_info_from_rule(rule_type, rule),  # type:ignore
+                }
+            )
+
+        return formatted_rules  # type:ignore
+
+    def _extract_info_from_rule(
+        self, rule_type: RuleType, rule: Union[BaseRule, ReleaseRule]
+    ) -> Dict[str, Union[List[str], str, None]]:
+        if rule_type == RuleType.BOOST_LATEST_RELEASES_RULE:
+            return {
+                "release": rule["condition"]["inner"][0]["value"],  # type:ignore
+                "environment": rule["condition"]["inner"][1]["value"],  # type:ignore
+            }
+        elif rule_type == RuleType.BOOST_KEY_TRANSACTIONS_RULE:
+            return {"transaction": rule["condition"]["inner"][0]["value"]}  # type:ignore
```

Comment:
This simplistic implementation has been done in order to avoid complex logic for rule validation, under the assumption that the logging is updated with the rules schema. If this is not the case we just get an error and silently fail.